runtime/conditions/merge_test.go

```diff
@@ -179,6 +179,64 @@ func TestMergeRespectPriority(t *testing.T) {
 	}
 }
 
+func TestMergeRespectGenerations(t *testing.T) {
+	tests := []struct {
+		name               string
+		negativeConditions []string
+		conditions         []*metav1.Condition
+		mergeOpts          []MergeOption
+		want               *metav1.Condition
+	}{
+		{
+			name:               "no generations option",
+			negativeConditions: []string{true1.Type},
+			conditions: []*metav1.Condition{
+				conditionWithGeneration(false1, 1),
+				conditionWithGeneration(true1, 2),
+				conditionWithGeneration(unknown1, 3),
+			},
+			want: FalseCondition("foo", "reason true1", "message true1"),
+		},
+		{
+			name:               "one generation",
+			negativeConditions: []string{true1.Type},
+			conditions: []*metav1.Condition{
+				conditionWithGeneration(false1, 1),
+				conditionWithGeneration(true1, 2),
+				conditionWithGeneration(unknown1, 3),
+			},
+			mergeOpts: []MergeOption{WithGenerations(3)},
+			want:      UnknownCondition("foo", "reason unknown1", "message unknown1"),
+		},
+		{
+			name:               "multiple generations",
+			negativeConditions: []string{true1.Type},
+			conditions: []*metav1.Condition{
+				conditionWithGeneration(false1, 1),
+				conditionWithGeneration(true1, 2),
+				conditionWithGeneration(unknown1, 3),
+			},
+			mergeOpts: []MergeOption{WithGenerations(2, 3)},
+			want:      FalseCondition("foo", "reason true1", "message true1"),
+		},
+	}
```

Comment:
Think that if there are multiple generations, the latest (highest) generation should always take precedence as this is the most up-to-date information available.

Reply:
Yeah, that'd be more correct. I'll implement that. Thanks.